src/api/app/controllers/webui/request_controller.rb

```diff
@@ -318,7 +298,10 @@ def forward_request_to(fwd)
   def set_package
     return unless params.key?(:package_name)
 
-    @package = Package.find_by(name: params[:package_name])
+    @package = Package.get_by_project_and_name(@project.to_param, params[:package_name],
```

Comment:
Good catch! As discussed with @DavidKang, I'm going to push this in a separate PR to have it faster on master since it's not directly related to this PR.